src/Chainweb/CutDB.hs

```diff
@@ -495,10 +494,10 @@ processCuts conf logFun headerStore payloadStore cutHashesStore queue cutVar = q
 
     hdrStore = _webBlockHeaderStoreCas headerStore
 
-    graph = _chainGraph headerStore
-
-    threshold :: Int
-    threshold = int $ 2 * diameter graph * order graph
+    threshold :: Cut -> Int
+    threshold c = int $ 2 * diameter graph * order graph
+      where
+        graph = chainGraphAt_ headerStore (meanChainHeight c)
```

Comment:
It doesn't really matter here, since it's a heuristics and it only makes a difference during a transition. But given how the new transition algorithm works, minimum height would actually make more sense I think. I'll change it.